### ISSUE TITLE
warning messages when compiling prototype

### ISSUE THREAD

**Issue by hppritcha** [hpc/ompi#41]
Seeing some compiler warnings when building the rebased sessions_new branch:
```
coll_adapt_module.c: In function 'adapt_module_enable':
coll_adapt_module.c:94:33: warning: format '%d' expects argument of type 'int', but argument 3 has type 'ompi_comm_extended_cid_t' {aka 'struct ompi_comm_extended_cid_t'} [-Wformat=]
   94 |                                 "(%d/%s): no underlying " # __api"; disqualifying myself", \
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~
   95 |                                 comm->c_contextid, comm->c_name); \
      |                                 ~~~~~~~~~~~~~~~~~
      |                                     |
      |                                     ompi_comm_extended_cid_t {aka struct ompi_comm_extended_cid_t}
../../../../opal/util/output.h:410:36: note: in definition of macro 'opal_output_verbose'
  410 |             opal_output(output_id, __VA_ARGS__);                     \
      |                                    ^~~~~~~~~~~
coll_adapt_module.c:110:5: note: in expansion of macro 'ADAPT_SAVE_PREV_COLL_API'
  110 |     ADAPT_SAVE_PREV_COLL_API(reduce);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
coll_adapt_module.c:94:33: warning: format '%d' expects argument of type 'int', but argument 3 has type 'ompi_comm_extended_cid_t' {aka 'struct ompi_comm_extended_cid_t'} [-Wformat=]
   94 |                                 "(%d/%s): no underlying " # __api"; disqualifying myself", \
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~
   95 |                                 comm->c_contextid, comm->c_name); \
      |                                 ~~~~~~~~~~~~~~~~~
      |                                     |
      |                                     ompi_comm_extended_cid_t {aka struct ompi_comm_extended_cid_t}
../../../../opal/util/output.h:410:36: note: in definition of macro 'opal_output_verbose'
  410 |             opal_output(output_id, __VA_ARGS__);                     \
      |                                    ^~~~~~~~~~~
```
These messages appear when compiling the ```adapt``` and ```han``` collectives files.

**Comment by hppritcha** [comment]
@tomhers could you take a look at these compiler warnings?

**Comment by hppritcha** [comment]
closed via 25fe776b
